src/components/portfolio/DetailsMainPositions.tsx

```diff
@@ -32,171 +16,48 @@ const useStyles = makeStyles(({ palette }: Theme) =>
-                        paintJob
-                        doLimit
-                      />
-                    }
-                  />
-                  <ValueOverName
-                    value={
-                      <StyledNumberFormat
-                        value={p.stock.perf1y}
-                        suffix="%"
-                        paintJob
-                      />
-                    }
-                    name={t('portfolio.details.year')}
-                    secondValue={
-                      <StyledNumberFormat
-                        value={oneYearAbsolute}
-                        suffix="€"
-                        paintJob
-                        doLimit
-                      />
-                    }
-                  />
-                  <ValueOverName
-                    value={
-                      <StyledNumberFormat
-                        value={p.totalReturnPercent}
-                        suffix="%"
-                        paintJob
-                      />
-                    }
-                    name={t('portfolio.details.totalReturn')}
-                    secondValue={
-                      <StyledNumberFormat
-                        value={p.totalReturn}
-                        suffix="€"
-                        paintJob
-                        doLimit
-                      />
-                    }
-                  />
-                </div>
-              </CardContent>
-              <CardActions className={classes.cardAction}>
-                <Button
-                  size="small"
-                  endIcon={
-                    <ArrowRightIcon
-                      style={{ marginLeft: '-8px', marginTop: '4px' }}
-                      aria-label="rightArrow"
-                    />
-                  }
-                  className={classes.button}
-                  style={{ backgroundColor: 'transparent' }}
-                  onClick={() => stockDetails(p.stock.symbol)}
-                >
-                  {t('portfolio.details.viewMore')}
-                </Button>
-              </CardActions>
-            </Card>
-          </GridListTile>
-        ))}
-      </GridList>
-    </div>
+    <>
+      <div className={classes.gridListWrapper}>
+        <GridList
+          cellHeight="auto"
+          cols={4}
+          spacing={32}
+          className={classes.gridList}
+          style={{ margin: '0 auto' }}
+        >
+          {positions.length <= 8 &&
+            positions.map((p) => (
+              <GridListTile key={p.stock.symbol}>
+                <DetailsMainPositionsCard p={p} />
+              </GridListTile>
+            ))}
+          {positions.length > 8 &&
+            positions.slice(0, 8).map((p) => (
+              <GridListTile key={p.stock.symbol}>
+                <DetailsMainPositionsCard p={p} />
+              </GridListTile>
+            ))}
+        </GridList>
+      </div>
+      {positions.length > 8 && (
```

Comment:
Couldn't you put this into the conditional render above? Both have the condition positions.length > 8

Reply:
Unfortunately due to styling, I can't put it in the statement above. It would mess up the show more bar, which would be displayed as another card in the grid...